src/Language/JVM/Utils.hs

```diff
@@ -117,20 +117,19 @@ instance (Binary w, Integral w) => Binary (SizedByteString w) where
     put (byteStringSize sbs)
     putByteString bs
 
-replaceJavaZeroWithNormalZero :: [Word8] -> [Word8]
-replaceJavaZeroWithNormalZero [] = []
-replaceJavaZeroWithNormalZero [h] = [h]
-replaceJavaZeroWithNormalZero (h1:h2:t) =
-  case (h1, h2) of
-    (192, 128) -> 0:(replaceJavaZeroWithNormalZero t)
-    _ -> h1:(replaceJavaZeroWithNormalZero $ h2:t)
-
-replaceNormalZeroWithJavaZero::[Word8] -> [Word8]
-replaceNormalZeroWithJavaZero [] = []
-replaceNormalZeroWithJavaZero (h:t) =
-  case h of
-    0 -> 192:128:(replaceNormalZeroWithJavaZero t)
-    _ -> h:(replaceNormalZeroWithJavaZero t)
+replaceJavaZeroWithNormalZero :: BS.ByteString -> BS.ByteString
+replaceJavaZeroWithNormalZero bs =
```

Comment:
``
replaceJavaZeroWithNormalZero :: BS.ByteString -> BS.ByteString
replaceJavaZeroWithNormalZero = go
  where
    go bs = 
      case BS.breakSubstring "\192\128" bs of
         (h, "") -> h
         (h, t) -> h `BS.append` "\0" `BS.append` go (BS.drop 2 t)